### ISSUE TITLE
Editor/Linting Advice?

### ISSUE THREAD

**Issue by OliverEvans96** [dtolnay/cxx#684]
Hi,

I'm really appreciating this great package! The one thing that's giving me a headache is linting in my editor (I'm currently using VS Code). I can't seem to configure it correctly recognize the header files. I'm following the docs on [Cargo-based builds](https://cxx.rs/build/cargo.html). 

It seems like you're doing some very fancy magic to make all of the headers available at compile time, but it doesn't seem like any header files are actually being written to disk. Any advice here would be appreciated!

Thanks,
Oliver

**Comment by dtolnay** [comment]
They are written to disk. I don't use VS Code so I can't directly help, but you'd need to make sure the editor's/linter's C++ include path contains the directories that build.rs feeds to the C++ compiler invocation. `includePath` in https://code.visualstudio.com/docs/cpp/c-cpp-properties-schema-reference seems relevant. If you put a `panic!()` at the bottom of build.rs it'll print out the list of include directories under a heading "*CXX include path*".

**Comment by OliverEvans96** [comment]
Thanks! I've been struggling with VSCode includePath for C++ unrelated to cxx, but I think I just got it working using [`bear`](https://github.com/rizsotto/Bear) to generate `compile_commands.json` which VSCode will read to know the specific import paths for each file.

I just did `bear -- cargo build`, which produces `compile_commands.json` in the project root, then set `"compileCommands": "./compile_commands.json"` in my configuration in `.vscode/c_cpp_properties.json`. The squiggly lines have disappeared!

One concern, though, is that cxx seems to append a hash to the generated directory name, e.g. `target/debug/build/my-project-754f540b5e93b54c/out/cxxbridge/include`. I have multiple versions of this in my target directory. How can I know when a new one will be generated and I should re-run `bear -- cargo build`?
